mempool/mempool.go

```diff
@@ -831,18 +824,28 @@ func (mp *TxPool) maybeAcceptTransaction(tx *btcutil.Tx, isNew, rateLimit, rejec
 		txscript.StandardVerifyFlags, mp.cfg.SigCache)
 	if err != nil {
 		if cerr, ok := err.(blockchain.RuleError); ok {
-			return nil, chainRuleError(cerr)
+			return nil, nil, chainRuleError(cerr)
 		}
-		return nil, err
+		return nil, nil, err
+	}
+
+	txD := &TxDesc{
```

Comment:
I think this would be cleaner and less code movement to just have `addTransaction` return the TxDesc.

```Go
txD := mp.addTransaction(utxoView, tx, bestHeight, txFee)
```